packages/rum-core/src/domain/rumEventsCollection/view/trackViewMetrics.ts

```diff
@@ -31,10 +110,31 @@ export function trackViewMetrics(
     viewStart,
     (newLoadingTime) => {
       viewMetrics.loadingTime = newLoadingTime
+
+      // We compute scroll metrics at loading time to ensure we have scroll data when loading the view initially
+      // This is to ensure that we have the depth data even if the user didn't scroll or if the view is not scrollable.
+      if (isExperimentalFeatureEnabled(ExperimentalFeature.SCROLLMAP)) {
+        const { scrollHeight, scrollDepth, scrollTop } = computeScrollMetrics()
+
+        scrollMetrics = {
+          maxScrollHeight: scrollHeight,
+          maxScrollDepth: scrollDepth,
+          maxScrollDepthTime: newLoadingTime,
```

Comment:
I kind of explain this in my previous comment, I think it's not ideal because the page is often not fully loaded at "loading time" I'm open to suggestions on how to improve this